src/language/records.md

```diff
@@ -0,0 +1,160 @@
+---
+title: Records
+description: Summary of the record data structure in Dart.
+toc: false
+---
+
+Records are an anonymous, immutable, aggregate type. Like other [collection types][], 
+they let you bundle multiple objects into a single object. Unlike other collection 
+types, records are fixed-sized, heterogeneous, and typed.
+
+Records are real values; you can store them in variables, pass them to and from
+functions, and store them in lists.
+
+## Record syntax
+
+_Records expressions_ are comma-delimited lists of named or positional fields,
+enclosed in parenthesis:
+
+```dart
+var record = ('first', a: 2, b: true, 'last');    // Record expression
+```
+
+_Record type annotations_ are comma-delimited lists of types enclosed in parenthesis.
+The `(int, int)` statements here are record type annotations used as a return
+type and as parameter types:
+
+```dart
+(int, int) swap((int, int) record) {
+  var (a, b) = record;
+  return (b, a);
+}
+```
+
+Named fields inside _record type annotations_ go inside a brace-delimited section
+of type and name pairs, after all positional fields:
+
+```dart
+// Record type annotation in a variable declaration
+(String, String?, {int a, bool b}) record;
+```
+
+Naming fields in record type annotations is optional for documentation purposes.
+The names of those named fields are not optional, though,  
+once you create a record with named fields. The field names become part of the
+[record's type definition](#record-types). They can be omitted, but not replaced.
```

Comment:
Wording seems confusing, maybe more clearly differentiate what is the difference between a "field with a name" and "named field"? Maybe mention that the first one concerned "positional fields"? Bob's analogy to function parameters seems good.

Reply:
Changed 43-45 to:

> Once you create a record using a record expression, though,
> any field names become part of the [record's type definition](#record-types)
> and are immutable. They can be omitted, but not replaced.

And then the link to the record equality section shows the example:

> ```dart
> // point =/= color
> ({int x, int y, int z}) point = (x: 1, y: 2, z: 3);
> ({int r, int g, int b}) color = (r: 1, g: 2, b: 3);
> ```


 is that better? Admittedly, I've been having trouble capturing this point, so any feedback helps

Reply:
I'll mark this as resolved since you adopted munificent's suggestion